File: src/main/java/com/EduTrack/EDUTRACK_Portfolio_MS/model/AcademicYear.java
package com.EduTrack.EDUTRACK_Portfolio_MS.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AcademicYear {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String year;
    private LocalDate startDate;
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    private AcademicYearStatus status;

    public enum AcademicYearStatus {
        ACTIVE, CLOSED, PLANNED
    }

}

File: frontend/src/app/core/models/academic-year.ts
export interface AcademicYear {
  id?: number;
  endDate: Date;
  status: AcademicYearStatus;
  year: string;
  startDate: Date;
  canEqual?: boolean;
  hashCode?: boolean;
  toString?: boolean;
}

export interface AcademicYearBuilder {
  id: number;
  year: string;
  startDate: Date;
  endDate: Date;
  status: AcademicYearStatus;
}

export enum AcademicYearStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  UPCOMING = 'UPCOMING',
  COMPLETED = 'COMPLETED'
}